client/src/modules/email-report/email-report.ctrl.js

```diff
@@ -0,0 +1,230 @@
+ */
+
+function EmailReportController($state, EmailReportSvc, ReportGroupSvc
+  , Session, util, Notify, ScrollTo, bhConstants, uiGridConstants) {
+
+  var vm = this;
+  /*
+   emailReport is the object
+   that contains : name,email, frequence and report-group code in UI form
+  */
+  vm.emailReport = {};
+  vm.selectedEmailReport = { selected : false };
+  vm.save = save;
+  vm.remove = remove;
+  vm.frequencies = [];
+  /*
+    when saving the profile, the user should select a report group
+    in orther to specify what report this email will get
+    the bellow array will contains those report
+  */
+  vm.report_groups = [];
+
+  init();
+
+  /*
+   initialise importants values for the UI form
+   such as the frequencies and report groups
+   that should be loaded from ReportGroupService
+
+   */
+
+
+  function init() {
+    /* loading frequencies*/
+    vm.frequencies = EmailReportSvc.frequencies;
+
+    /* If the user has selected a profile, fill name and email field */
+    try {
+      if ($state.params.data.email) {
+        vm.emailReport.email = $state.params.data.email;
+        vm.emailReport.name = $state.params.data.name;
+      }
+    } catch (ex) {
+        // exception
+    }
+
+    // IF the user want to edit an email report, action->edit
+    try {
+
+      if ($state.params.data.selectedEmailReport.id) {
+        vm.emailReport = $state.params.data.selectedEmailReport;
+        vm.selectedEmailReport = vm.emailReport;
+        vm.selectedEmailReport.selected = true;
+      }
+    } catch (ex) {
+      // execption
+    }
+
+
+    // loading report groups
+    ReportGroupSvc.read().then(function (reportGroups) {
+      vm.report_groups = reportGroups;
+    })
+      .catch(Notify.handleError);
+  }
+
+  vm.getFrequencyLabel = function (key) {
+
+    for (let i = 0; i < vm.frequencies.length; i++) {
+      const fq = vm.frequencies[i];
+      if (fq.key === key) {
+        return fq.label;
+      }
+    }
+
+  }
+
+
+  function confirmDeleteProfile() {
+
+  }
+  // reset the form state
+  function resetForm(RegistrationForm) {
+
+    vm.emailReport = {};
+    vm.selectedEmailReport = {};
+    vm.selectedEmailReport.selected = false;
+
+    RegistrationForm.$setPristine();
+    RegistrationForm.$setUntouched();
+    ScrollTo('anchor');
+  }
+
+
+  // inserting a new email for a  reporting in the database
+  function save(RegistrationForm) {
+
+    // end propagation for invalid state - this could scroll to an $invalid element on the form
+    if (RegistrationForm.$invalid) {
+      return Notify.danger('FORM.ERRORS.INVALID');
+    }
+
+    const isCreateReport = vm.selectedEmailReport.selected === false;
+
+    let operation;
+
+    if (isCreateReport) {
+      operation = EmailReportSvc.create(vm.emailReport);
+    } else {
+      operation = EmailReportSvc.update(vm.emailReport);
+    }
+
+    operation.then(function (confirmation) {
+
+      if (confirmation) {
+        alert('operation successeded');
```

Comment:
Please do not use `alert()` on the client.  A better option is to use [ModalService](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/ModalService.js#L35)'s alert() method instead.

[This was previously mentioned in a review](https://github.com/IMA-WorldHealth/bhima-2.X/pull/1778#discussion_r130851350).